storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -136,9 +136,15 @@ private Store(
     this.justified_checkpoint = justified_checkpoint;
     this.best_justified_checkpoint = best_justified_checkpoint;
     this.blocks = blocks;
-    this.votes = new HashMap<>(votes);
     this.highestVotedValidatorIndex =
-        this.votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO);
+        votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO);
+    this.votes =
+        new VoteTracker
+            [Math.max(this.highestVotedValidatorIndex.intValue() + 1, DEFAULT_VOTE_TRACKER_SIZE)];
```

Comment:
We're aiming to leave some extra space after the highest current validator index so we don't have to resize the array again too soon, so:

```suggestion
        new VoteTracker[this.highestVotedValidatorIndex.intValue() + DEFAULT_VOTE_TRACKER_SIZE)];
```

And I'd probably rename `DEFAULT_VOTE_TRACKER_SIZE` to something like `VOTE_TRACKER_SPARE_CAPACITY`

Reply:
done,  I slightly missed the spare capacity logic in Store.constructor 😅